test/colors/colors.test.js

```diff
@@ -43,6 +43,25 @@ describe('colorts', () => {
         expect(exitCode).toBe(0);
     });
 
+    it('should disable colored output with --no-color', () => {
+        const { stderr, stdout, exitCode } = run(__dirname, ['--stats=verbose', '--no-color']);
+
+        expect(stderr).toBeFalsy();
+        const output = isWebpack5 ? 'successfully' : 'main.js';
+        expect(stdout).not.toContain(`\u001b[1m\u001b[32m${output}\u001b[39m\u001b[22m`);
+        expect(stdout).toContain(output);
+        expect(exitCode).toBe(0);
+    });
+
+    it('should work with the "stats" option and --color flags', () => {
+        const { stderr, stdout, exitCode } = run(__dirname, ['--stats=verbose', '--color']);
+
+        expect(stderr).toBeFalsy();
+        const output = isWebpack5 ? 'successfully' : 'main.js';
+        expect(stdout).toContain(`\u001b[1m\u001b[32m${output}\u001b[39m\u001b[22m`);
+        expect(exitCode).toBe(0);
+    });
```

Comment:
Let's add couple test more:
- `stats: { colors: true }` and `--no-color`
- `stats: { colors: false }` and `--color`

Reply:
I think we prefer colors in config over --color. 
https://github.com/webpack/webpack-cli/blob/cd567127fe3651a9a7d0bc21cf79ae4a78c0e3c4/packages/webpack-cli/lib/webpack-cli.js#L370

Reply:
nope flag should have preference

Reply:
Yes, but let's add tests to avoid regressions

Reply:
I think we should prefer `--color` over `stats.colors` in this case

Reply:
Because I want to disable colors when run `--no-color`

Reply:
And using `coloretteOptions.enabled` by default if no `stats.color` or `--color`/`--no-color`

Reply:
```
stats.colors = typeof stats.colors !== 'undefined' &&  !args.colors ? stats.colors : coloretteOptions.enabled
```

Reply:
```js
let colors;

// From flags
if (typeof args.colors !== 'undefined') { 
  colors = args.colors;
} 
// From stats
else if (typeof stats.colors !== 'undefined') {
   colors = stats.colors;
} 
// Default
else {
 colors = coloretteOptions.enabled;
}
```

Reply:
done 💯